goaviatrix/vpn_user.go

```diff
@@ -147,3 +148,15 @@ func (c *Client) DeleteVPNUser(vpnUser *VPNUser) error {
 	}
 	return nil
 }
+
+func CheckVpnUserSettings(vpnUser *VPNUser) error {
+	if vpnUser.DnsName == "" && vpnUser.VpcID == "" && vpnUser.GwName == "" {
+		return fmt.Errorf("please set 'vpc_id' and 'gw_name', or 'dns_name' alone")
+	} else if vpnUser.DnsName == "" && (vpnUser.VpcID == "" || vpnUser.GwName == "") {
+		return fmt.Errorf("please set both 'vpc_id' and 'gw_name'")
+	} else if vpnUser.DnsName != "" && (vpnUser.VpcID != "" || vpnUser.GwName != "") {
+		return fmt.Errorf("DNS is enabled. Please set 'vpc_id' and 'gw_name' to be empty")
+	}
```

Comment:
I didnt mean to put into a function
I meant

if vpnUser.DnsName == "" {
	if vpnUser.VpcID == "" && vpnUser.GwName == "" {
		return fmt.Errorf("please set 'vpc_id' and 'gw_name', or 'dns_name' alone")
	} else if (vpnUser.VpcID == "" || vpnUser.GwName == "") {
		return fmt.Errorf("please set both 'vpc_id' and 'gw_name'")
	}
}
}else {
	if vpnUser.VpcID != "" || vpnUser.GwName != "") {
		return fmt.Errorf("DNS is enabled. Please set 'vpc_id' and 'gw_name' to be empty")
	}
}